### PR TITLE
Add List and Details Page for Volume Snapshot

### PR DESCRIPTION
**Pull request by bipuladh**
- Details Page
- List Page
- Filter based on Status
Screenshot from 2020-07-07 17-44-02
Screenshot from 2020-07-07 17-43-53

### CODE REVIEW

File: frontend/public/components/factory/table.tsx [hunk 1]
```diff
@@ -145,6 +146,7 @@ const sorts = {
   machinePhase: (machine: MachineKind): string => getMachinePhase(machine),
   nodePods: (node: NodeKind): number => nodePods(node),
   pvcUsed: (pvc: K8sResourceKind): number => pvcUsed(pvc),
+  volumeSnapshotSize: (snapshot: K8sResourceKind): number => snapshotSize(snapshot),
```
Comment (rawagner) [inline frontend/public/components/factory/table.tsx:149]: seems like this should be 
```suggestion
  volumeSnapshotSize: (snapshot: VolumeSnapshotKind): number => snapshotSize(snapshot),
```

File: frontend/packages/console-shared/src/sorts/snapshot.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { convertToBaseValue } from '@console/internal/components/utils';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+export const snapshotSize = (snapshot: K8sResourceKind): number => {
```
Comment (rawagner) [inline frontend/packages/console-shared/src/sorts/snapshot.ts:4]: we can be more specific, right ?
```suggestion
export const snapshotSize = (snapshot: VolumeSnapshotKind): number => {
```

Comment (bipuladh) [reply frontend/packages/console-shared/src/sorts/snapshot.ts]: all day every day

**Comment by bipuladh** [comment]
cc @spadgett we are adding a new tab under storage.

**Pull request merged**